common/src/main/java/com/wynntils/services/mapdata/providers/json/JsonProvider.java

```diff
@@ -285,13 +284,21 @@ public JsonElement serialize(MapAttributesImpl mapAttributes, Type type, JsonSer
     }
 
     public static final class JsonIconSerializer implements JsonDeserializer<MapIconImpl>, JsonSerializer<MapIconImpl> {
+        private static final String DATA_URL_PREFIX = "data:image/png;base64,";
+
         @Override
         public MapIconImpl deserialize(JsonElement jsonElement, Type jsonType, JsonDeserializationContext context)
                 throws JsonParseException {
             JsonObject json = jsonElement.getAsJsonObject();
 
             String id = json.get("id").getAsString();
             String base64Texture = json.get("texture").getAsString();
+
+            // Remove the data url prefix if it exists
+            if (base64Texture.startsWith(DATA_URL_PREFIX)) {
+                base64Texture = base64Texture.substring(DATA_URL_PREFIX.length());
```

Comment:
I don't think we need this ambiguity in the "spec". Either we say it is a URL and we support `"data:image/png;base64`, or we stipulate that it must be a base-64-encoded png. I tend towards the latter, it probably shuts the door completely for some possible security holes.